### ISSUE TITLE
HEAVY memory usage (>8GB)

### ISSUE THREAD

**Issue by SEAPUNK** [tmux/tmux#706]
- tmux version

tmux 2.3

- Platform

Arch Linux

- Config:

The current config is:

```
set -g mouse on

# List of plugins
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-sensible'
set -g @plugin 'tmux-plugins/tmux-resurrect'

# Initialize TMUX plugin manager
run '~/.tmux/plugins/tpm/tpm'
```

But I'm pretty sure this happened with this config as well:

```
set -g mouse on
```

- Logs:

Can't at this time, especially due to how difficult it is to trigger it

- $TERM

Outside tmux: xterm

Inside tmux: screen-256color

---

I have about 8 tmux sessions running on a server, each session having about 4 to 8 panes each. I have another computer that is almost always on, ssh'd into the server, with those tmux sessions perpetually opened, so I can monitor what's going on in those sessions.

Normally, it all works fine. However, sometimes, when a tmux session pane (or panes) start writing to stdout/stderr at very high speeds (when a program is logging something very quickly, or logging something very large), it seems to trigger a bug where either the tmux session or the ssh session just locks up, forcing me to close the terminal tab, reconnect, and `tmux a -t session_name`.

When the ssh (or tmux?) session locks up, it seems to trigger a bug where tmux apparently seems to collect something to the point the memory usage grows about 6GB each time this happens.

It's not very easy for me to reproduce, because it only happens _sometimes_. Luck of the draw. Sometime today it appeared to have happened again, since the I just got an alert that tmux is eating up 17 GB. Killing tmux and restarting the sessions (in my case, with resurrect) fixes the problem until the lock-up happens again.

---

I'll build tmux from master, and try to reproduce this issue with logs the next time I get access to the computer that's having this problem.

**Comment by nicm** [comment]
Hmm, I don't know what is going on here, please make sure you can reproduce with master and if possible get some logs.

**Comment by SEAPUNK** [comment]
Alright, since the problem happened again, I built tmux from master, and running it in verbose mode: `tmux -vvvv -Ltest -f/dev/null new`

My worry here is that the log file will grow to an enormous size before the bug gets triggered again.

**Comment by spencera1** [comment]
I'm also experiencing this issue with 2.3 on Fedora. All I have to do is start the tmux process, and I can watch the memory usage slowly increase using pmap or top. No additional activity (no new windows, instances, etc.) is necessary to cause the increase. After about 3 days or so, the memory usage was >5GB.

**Comment by sharpevo** [comment]
@SEAPUNK I also installed the resurrect plugin, and most of time out of memory when saving the sessions.

**Comment by SEAPUNK** [comment]
IIRC, building the latest tmux from source fixed the problem for me, because I haven't had issues with it like that since. Maybe the binaries provided by the package manager compile with settings that cause issues?

**Comment by ivan** [comment]
This might have been fixed in 2.5 by https://github.com/tmux/tmux/blob/91b220525b0406763dafb6698d2741bec580bc10/CHANGES#L257-L260

**Comment by X41** [comment]
after updating to the newest arch linux package (and setting the history limit to 300 lines) my ram useage is at a few megabytes. can't tell for sure if it was the update or the limit i've set, but i'm pretty sure it's the fixed bug
